server/src/main/java/org/elasticsearch/transport/OutboundHandler.java

```diff
@@ -199,6 +209,7 @@ private void maybeLogSlowMessage(boolean success) {
                     final long logThreshold = slowLogThresholdMs;
                     if (logThreshold > 0) {
                         final long took = threadPool.relativeTimeInMillis() - startTime;
+                        handlingTimeTracker.addHandlingTime(took);
```

Comment:
This may be spurious since it counts time spent waiting for the channel to become writeable (cf https://github.com/elastic/elasticsearch/issues/77838). Should we track it separately from the inbound time tracking?

Reply:
I don't think it's going to be possible to differentiate between waiting for writable and time spent actually writing really (I know I promised differently a while back sorry about that). Definitely not easily. If you think about it, you could run into a non-writable channel and start counting from there. But then once it becomes writable again, you might not be the first in line to get your bytes flushed because some other write comes before you and may turn the channel not-writable again. And then that other write takes CPU for TLS and such, making it really hard to cleanly define what time was spent waiting.

I really like the current number for the simple fact that it gives an indication for the latency overall on a transport thread (while the inbound handler check indicates individual per message slowness). I don't really see how we could cleanly identify the fact that a channel isn't writable for an extended period of time and pinpoint that on the network.

Reply:
> . Should we have two histograms, one for inbound things (which is purely handling-time) and one for outbound things (which potentially includes channel-blocked time)?

Yea that would be optimal actually. Can we do that here? :)

Reply:
:+1: done in c775e4e.

Another related observation is that we're not tracking outbound time for HTTP responses AFAICT. Should we? Can we?

Reply:
> Should we? Can we?

We could. I guess it would be nice to have but probably not all that valuable. The distribution on the outbound side for HTTP will be the same as that for sending transport messages. For REST outbound I'm almost thinking I'd rather like the distribution of serialization times there, because those have historically been the problem and it would give us more information when going on the hunt for why we have a bad distribution of outbound handling times.
IMO that'd be a worthwhile follow-up.